modules/transaction-payment/src/lib.rs

```diff
@@ -293,6 +312,19 @@ pub mod module {
 	pub enum Error<T> {
 		/// The swap path is invalid
 		InvalidSwapPath,
+		/// Supply currency balance too low
+		SupplyBalanceTooLow,
+		/// Asset no in the charge fee pool
+		AssetNotInFeePool,
+	}
+
+	#[pallet::event]
+	#[pallet::generate_deposit(pub(crate) fn deposit_event)]
+	pub enum Event<T: Config> {
+		TriggerThresholdUpdated(Balance),
+		TriggerSwap(Balance, Balance, CurrencyId),
+		SwapFromDex(Balance, Balance, Ratio),
+		ChargeFee(Balance, Balance, Ratio),
```

Comment:
1. Add event dostrings like

```rust
/// Trigger threshold updated. \[Balance, Balance\]
TriggerThresholdUpdated(Balance),
```

2. Rename events. In general, events naming convention is like:
- DO: `FeeCharged`
- DO NOT: `ChargeFee`

Reply:
Actually we should just use named fields for new events. i.e. `TriggerThresholdUpdated { new_value: Balance }`

Reply:
As the `transfer` action already produce events. do we need `FeeCharged` event that show the supply asset balance transfer to and native asset got from treasury account, also the rate currently used?